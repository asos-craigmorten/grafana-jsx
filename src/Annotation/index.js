/**
 * REF: https://grafana.com/docs/grafana/latest/reference/dashboard/#annotations
 */

import PropTypes from "prop-types";

const Annotation = ({
  builtIn = 1,
  datasource = "-- Grafana --",
  enable = true,
  hide = true,
  iconColor = "rgba(0, 211, 255, 1)",
  name = "Annotations & Alerts",
  type = "dashboard",
  ...props
}) => (
  <annotation
    builtIn={builtIn}
    datasource={datasource}
    enable={enable}
    hide={hide}
    iconColor={iconColor}
    name={name}
    type={type}
    {...props}
  />
);

Annotation.PropTypes = {
  builtIn: PropTypes.oneOf([0, 1]),
};

export default Annotation;
