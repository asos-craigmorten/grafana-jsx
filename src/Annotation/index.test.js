import Annotation from "./";

const withAllProps = {
  builtIn: 0,
  datasource: "test-datasource",
  enable: false,
  hide: false,
  iconColor: "test-colour",
  name: "test-name",
  type: "test-type",
};

describe("Annotation", () => {
  describe("when passed no props", () => {
    it("should render a default Annotation", () => {
      expect(<Annotation />).toMatchSnapshot();
    });
  });

  describe("when passed all props", () => {
    it("should render a custom Annotation", () => {
      expect(<Annotation {...withAllProps} />).toMatchSnapshot();
    });
  });

  describe("when passed additional props", () => {
    it("should render a custom Annotation with the additional props", () => {
      expect(
        <Annotation
          {...withAllProps}
          test-annotation-prop={Symbol("test-annotation-value")}
        />
      ).toMatchSnapshot();
    });
  });
});
