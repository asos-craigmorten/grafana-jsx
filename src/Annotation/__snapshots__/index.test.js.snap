// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Annotation when passed additional props should render a custom Annotation with the additional props 1`] = `
Object {
  "builtIn": 0,
  "datasource": "test-datasource",
  "enable": false,
  "hide": false,
  "iconColor": "test-colour",
  "name": "test-name",
  "test-annotation-prop": Symbol(test-annotation-value),
  "type": "test-type",
}
`;

exports[`Annotation when passed all props should render a custom Annotation 1`] = `
Object {
  "builtIn": 0,
  "datasource": "test-datasource",
  "enable": false,
  "hide": false,
  "iconColor": "test-colour",
  "name": "test-name",
  "type": "test-type",
}
`;

exports[`Annotation when passed no props should render a default Annotation 1`] = `
Object {
  "builtIn": 1,
  "datasource": "-- Grafana --",
  "enable": true,
  "hide": true,
  "iconColor": "rgba(0, 211, 255, 1)",
  "name": "Annotations & Alerts",
  "type": "dashboard",
}
`;
